### PR TITLE
CB-15335 Introduce ClassValue as a representation of classes in database. Thi…

### PR DESCRIPTION
**Pull request by foldik**
…s fixes the issue when during a rollback some class disappears from the application the flow engine can handle it gracefully.

See detailed description in the commit message.

### CODE REVIEW

**Comment by lajosrodek** [comment]
Sorry, could you please add the relevant Jira ticket number if available?

File: flow/src/main/java/com/sequenceiq/flow/core/Flow2Handler.java [hunk 1]
```diff
@@ -444,43 +445,49 @@ public void restartFlow(String flowId) {
...
-                                flowLog.getOperationType().name(), span.context()), flowLog.getFlowChainId(), flowLog.getNextEvent(), payload);
-                        return;
-                    }
-                } catch (RuntimeException e) {
-                    String message = String.format("Flow could not be restarted with id: '%s', flow chain id: '%s' and flow type: '%s'", flowLog.getFlowId(),
-                            flowLog.getFlowChainId(), flowLog.getFlowType().getSimpleName());
-                    LOGGER.error(message, e);
-                }
-            }
+        checkArgument(flowLog.getFlowType() != null, "Can't restart fow %s because flow type is null.", flowLog.getFlowId());
```
Comment (lnardai) [inline flow/src/main/java/com/sequenceiq/flow/core/Flow2Handler.java:448]: typo, *flow

File: flow/src/main/java/com/sequenceiq/flow/core/Flow2Handler.java [hunk 2]
```diff
@@ -444,43 +445,49 @@ public void restartFlow(String flowId) {
...
+        checkArgument(flowLog.getFlowType().isOnClassPath(),
+                "Can't restart fow %s because flow type %s is not on classpath.", flowLog.getFlowId(), flowLog.getFlowType().getName());
```
Comment (lnardai) [inline flow/src/main/java/com/sequenceiq/flow/core/Flow2Handler.java:450]: These won't get into statusReason field right?

**Review by lnardai — APPROVED** [review]
Left 2 comments otherwise LGTM

**Pull request merged**